assembly/src/ast/parsers/context.rs

```diff
@@ -14,6 +15,7 @@ use vm_core::utils::{
 /// AST Parser context that holds internal state to generate correct ASTs.
 pub struct ParserContext<'a> {
     pub imports: &'a BTreeMap<String, LibraryPath>,
+    pub used_imported_procs: &'a mut ImportedProcsMap,
```

Comment:
We can probably use `ModuleImports` here. The reason I didn't was that I ran into some mutability issues when I first tried it that way, but now that I look at it again I can't see why it wouldn't. I'll refactor it.

Reply:
It doesn't seem to work. Take a look at the latest commit.

The problem is that `imports` is available when the `ParserContext` instance is created, whereas `used_imported_procs` is populated during parsing using `imports`. This means that `imports` must be marked as immutable, whereas `used_imported_procs` must be marked as mutable, and I can't see how we can achieve that without marking all of `ModuleImports` as mutable in the AST, which introduces lifetime problems.

Maybe I'm missing something, but off the top of my head I can't see an easy way to fix this, except to not use `ModuleImports` in `ParserContext`.

Reply:
The way to fix this is to move the logic which looks up module path and builds procedure ID into the `ModuleImports` struct (and I actually think that this is a slightly cleaner way to do it). I pushed a commit which shows how this can be done.

Reply:
Ah, got it. And yes, it's cleaner this way.